include/arch/arm/arch/machine.h

```diff
@@ -42,12 +42,10 @@ void cleanInvalidate_L1D(void);
 void cleanCaches_PoU(void);
 void cleanInvalidateL1Caches(void);
 
-/* Cleaning memory before user-level access */
+/* Cleaning memory before user-level access. Does not flush cache. */
 static inline void clearMemory(word_t *ptr, word_t bits)
```

Comment:
It seems the architecture specific `clearMemory()` is now just a wrapper around `memzero()` everywhere. We could drop it completely then to simplify things?

Reply:
I would prefer not to. `clearMemory()` is a machine interface function in the proofs and things have changed underneath a few times in the past. While it'd be nice if this was the last time, I doubt that this will be the case.

We could check back in a few years and if it has turned out to be stable, we can still move the interface.